stdlib/public/core/FloatingPointParsing.swift.gyb

```diff
@@ -256,15 +256,21 @@ extension ${Self}: LosslessStringConvertible {
   // Use the all-Swift `parse_float${bits}()` implementation for Float16/32/64
   @available(SwiftStdlib 5.3, *)
   public init?(_ text: Substring) {
-    // TODO: Someday, this whole function should simplify down to just:
-    // ${Self}(text.utf8.span)
   #if _pointerBitWidth(_16)
     // Always fail on 16-bit targets
     return nil
   #else
-    // Work around span availability limits
-    let parsed = unsafe text.base._guts.withFastUTF8 { chars -> ${Self}? in
-      unsafe parse_float${bits}(chars.span)
+    let parsed: ${Self}?
+    if text.base._guts.isFastUTF8 {
+      parsed = unsafe text.base._guts.withFastUTF8 { chars -> ${Self}? in
+        unsafe parse_float${bits}(chars.span.extracting(text._offsetRange))
+      }
+    } else {
+      // Otherwise, make a copy so we have contiguous UTF8...
+      let string = String(text)
+      parsed = unsafe string._guts.withFastUTF8 { chars -> ${Self}? in
```

Comment:
```suggestion
      var string = String(text)
      parsed = string.withUTF8 { chars -> ${Self}? in
```
If `text` is a substring of a bridged string that has the same range of indices as its base string (e.g. `let text = bridgedString[...]`), then the return string will still be a bridged string, and that wouldn't support `withFastUTF8`.